File: src/main/java/org/activecraft/activecraftcore/commands/KickAllCommand.java
package org.activecraft.activecraftcore.commands;

import org.activecraft.activecraftcore.ActiveCraftPlugin;
import org.activecraft.activecraftcore.exceptions.ActiveCraftException;
import org.activecraft.activecraftcore.messages.MessageFormatter;
import org.activecraft.activecraftcore.utils.ColorUtils;
import org.activecraft.activecraftcore.ActiveCraftPlugin;
import org.activecraft.activecraftcore.exceptions.ActiveCraftException;
import org.activecraft.activecraftcore.messages.MessageFormatter;
import org.activecraft.activecraftcore.utils.ColorUtils;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;

import java.util.List;

public class KickAllCommand extends ActiveCraftCommand {

    public KickAllCommand(ActiveCraftPlugin plugin) {
        super("kickall",  plugin);
    }

    @Override
    public void runCommand(CommandSender sender, Command command, String label, String[] args) throws ActiveCraftException {
        checkPermission(sender);
        MessageFormatter msgFormatter = new MessageFormatter(ChatColor.GOLD, "reason", ColorUtils.replaceColorAndFormat(concatArray(args)));
        sendMessage(sender, this.cmdMsg(args.length == 0 ? "default" : "custom", msgFormatter));
        Bukkit.getOnlinePlayers().stream()
                .filter(player -> !player.hasPermission("activecraft.kickall.bypass"))
                .forEach(player -> player.kickPlayer(
                        this.cmdMsg((args.length == 0 ? "default" : "custom") + "-message", msgFormatter)));
    }
    
    @Override
    public List<String> onTab(CommandSender sender, Command command, String label, String[] args) {
        return null;
    }
}

File: src/main/java/org/activecraft/activecraftcore/commands/KickAllCommand.kt
package org.activecraft.activecraftcore.commands

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.exceptions.ActiveCraftException
import org.activecraft.activecraftcore.utils.replaceColorAndFormat
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

class KickAllCommand(plugin: ActiveCraftPlugin) : ActiveCraftCommand("kickall", plugin) {
    @Throws(ActiveCraftException::class)
    public override fun runCommand(sender: CommandSender, command: Command, label: String, args: Array<String>) {
        assertCommandPermission(sender)
        val msgFormatter =
            newMessageFormatter().addFormatterPattern("reason", replaceColorAndFormat(joinArray(args)), ChatColor.GOLD)
        sendMessage(sender, this.cmdMsg(if (args.isEmpty()) "default" else "custom", msgFormatter))
        Bukkit.getOnlinePlayers()
            .filter { player -> !player.hasPermission("activecraft.kickall.bypass") }
            .forEach { player ->
                player.kickPlayer(this.cmdMsg((if (args.isEmpty()) "default" else "custom") + "-message", msgFormatter))
            }
    }

    public override fun onTab(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ) = null

}